src/Corporation/ui/IndustryDescription.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+
+import Typography from "@mui/material/Typography";
+
+import { MoneyCost } from "./MoneyCost";
+import { Corporation } from "../Corporation";
+import { IndustryType } from "../data/Enums";
+import { IndustriesData } from "../data/IndustryData";
+
+interface IProps {
```

Comment:
me too - I would never name it like that in my project. The only reason for this code style - is consistency in this project.

The way I prefer to implement things in react is:
1. have a directory named as your component (e.g. `IndustryDescription`)
2. add at least 3 files to this directory: `IndustryDescription.tsx`, IndustryDescription.types.ts` and `index.ts` exporting everything from previous files (barreling). In case of MUI styles there would also be a `IndustryDescription.styles.ts` which exports `useStyles` hook.
3. `IndustryDescription.types.ts` exports a type (not interface) named `IndustryDescriptionProps`
4. `IndustryDescription.tsx` imports `IndustryDescriptionProps` from `IndustryDescription.types.ts` exports component named `IndustryDescription`:
```tsx
import { IndustryDescriptionProps } from './IndustryDescription.types.ts';

const IndustryDescription = (props: IndustryDescriptionProps) => {
  ...
}
export { IndustryDescription } // export always in the end of file, never export default
```

However, applying my best practices in this project won't make sense if I do this only for a couple of places.
IMHO either I follow the current code style in the project or refactor everything (after agreement) and use a better way

Reply:
btw `interface IProps` is used 201 times in this project 
lol

Reply:
I would love to restructure everything about react in the way I explained above. I would fee like home then xD

Reply:
Yeah before I started doing work on the game, these IProps or ITypeName were all over the place. There are a lot less IProps and similar now, but there are still some of them out there. I didn't go find and rename all of them because it's not a big deal and has no actual impact, usually I just change them when I come across them. Like I said it's not really an issue

Reply:
I could create a PR (or series of PR) to change this pattern if this is something that you also want (to me current pattern looks ugly as well)

Reply:
Three's also the middle ground (that I think Snarling was suggesting): Fix these as you come across them, without doing a whole-project refactoring. Although if you have the energy for the latter, I guess go wild.

Reply:
@d0sboots @Snarling 
which of the following styles do you prefer for react component props:

1. Inline props
```tsx
export const AchievementEntry: VFC<{
  achievement: Achievement;
  unlockedOn?: number;
  cssFiltersUnlocked: string;
  cssFiltersLocked: string;
}> = ({ achievement, unlockedOn, cssFiltersUnlocked, cssFiltersLocked }) => {
```
2. Props in the same file
```tsx
type AchievementEntryProps = {
  achievement: Achievement;
  unlockedOn?: number;
  cssFiltersUnlocked: string;
  cssFiltersLocked: string;
};

export const AchievementEntry: VFC<AchievementEntryProps> = ({ achievement, unlockedOn, cssFiltersUnlocked, cssFiltersLocked }) => {...
```
3. Same as 2 but props are stored in a separate file, e.g. `AchievementEntry.types.ts`. In this case, each component will be nested into the same-name directory with props, component and an index file (for barreling)

And do you mind replacing a function declaration with an arrow function (like in the examples above)

Reply:
I think the answer is none of those?

I was confused about what this `VFC` thing was that you were doing, since I hadn't seen it in our codebase before. Turns out, it's deprecated: <https://www.mydatahack.com/moving-away-from-fc-and-vfc/>

So, I think the right answer is like 2, but don't export the type (there's no need), and don't do the VFC thing. 1-like would also be fine. *Definitely* not 3.

Reply:
So the first one without VFC should be fine then?
I don't see much value in declaring a type, using it on the next line, and never seeing it anywhere else again.
However, there are cases when you would like to reference a type of one of the props inside a component (i.e. `MyComponent['onChange']`) - in those cases having an explicit type declaration could make sense (even though you could easily use something like `typeof onChange` instead).

Reply:
@d0sboots BTW, the link you shared references react 18 (in which functional components don't have `children` prop) - but this project is based on react 17, so it could still be an option

Reply:
I took it to mean "`FC` and `VFC` are generally the way of the past, don't keep using them."